Translation: FAQ > Virtual DOM and Internals [Ready for review]

giossa94 opened the pull request:
The translation is ready for any feedback that you can provide @dmoralesm, @carburo, and @alejandronanez. 

I'm not really happy with these points:
- The title. I couldn't find a better translation for `Internals`, I thought on something like `DOM virtual y la implementación de React`, do you think that it is more accurate that the current one?
- I choose to kept `Shadow DOM` untranslated, because `DOM sombra` doesn't sound really good to me. On the other hand `Virtual DOM` is translated to `DOM Virtual`, which is a bit incosistent. Do you want to keep both of them untranslated or have any other idea for this? 

Relates to issue #4.

carburo commented on content/docs/faq-internals.md:
```diff
@@ -1,23 +1,23 @@
+### ¿Qué es el DOM virtual? {#what-is-the-virtual-dom}
 
-The virtual DOM (VDOM) is a programming concept where an ideal, or "virtual", representation of a UI is kept in memory and synced with the "real" DOM by a library such as ReactDOM. This process is called [reconciliation](/docs/reconciliation.html).
+El DOM virtual (VDOM) es un concepto de programación donde una representación ideal o "virtual" de la UI se mantiene en memoria y en sincronía con el DOM "real", mediante una biblioteca como ReactDOM. Este proceso se conoce como [reconciliación](/docs/reconciliation.html).
```
En aras de la consistencia estamos usando IU como sigla para interfaz de usuario.
```suggestion
El DOM virtual (VDOM) es un concepto de programación donde una representación ideal o "virtual" de la IU se mantiene en memoria y en sincronía con el DOM "real", mediante una biblioteca como ReactDOM. Este proceso se conoce como [reconciliación](/docs/reconciliation.html).
```

carburo commented on content/docs/faq-internals.md:
```diff
@@ -1,23 +1,23 @@
+El DOM virtual (VDOM) es un concepto de programación donde una representación ideal o "virtual" de la UI se mantiene en memoria y en sincronía con el DOM "real", mediante una biblioteca como ReactDOM. Este proceso se conoce como [reconciliación](/docs/reconciliation.html).
 
-This approach enables the declarative API of React: You tell React what state you want the UI to be in, and it makes sure the DOM matches that state. This abstracts out the attribute manipulation, event handling, and manual DOM updating that you would otherwise have to use to build your app.
+Este enfoque hace posible la API declarativa de React: le dices a React en qué estado quieres que esté la UI, y se hará cargo de llevar el DOM a ese estado. Esto abstrae la manipulación de atributos, manejo de eventos y actualización manual del DOM que de otra manera tendrías que usar para construir tu aplicación.
```
```suggestion
Este enfoque hace posible la API declarativa de React: le dices a React en qué estado quieres que esté la IU, y se hará cargo de llevar el DOM a ese estado. Esto abstrae la manipulación de atributos, manejo de eventos y actualización manual del DOM que de otra manera tendrías que usar para construir tu aplicación.
```

giossa94 commented:
> Very good, really good start! I just fixed a typo in «arbol -> árbol» and changed de use of «UI» for «IU» as it is the term we are using everywhere for consistency among articles. I didn't just merge because of the issue with the title. I made some suggestions. Tell me what you think.
> 
>     * Es cierto que el título no funciona bien tal y como está. Quizá podría ser «DOM virtual y elementos internos», «DOM virtual y detalles internos» o «DOM virtual y detalles de implementación». La última es la traducción más libre y quizá no encaja del todo en el espíritu del título.
> 
>     * Creo que mantener Shadow DOM no es un problema al ser una tecnología que no está asociada a React y que hasta ahora no se utiliza para ella una traducción (por ejemplo en MDN utilizan el término sin cambiar).

Thank you! :)

About your suggestions for the title, `DOM virtual y detalles de implementación` is the one that sounds most natural to me. I think that `detalles internos` or `elementos internos` are less common terms on spanish literature (docs/articles/books).

carburo commented:
Ok, very good! Ready to merge
